agent/envoyextensions/builtin/otel-access-logging/structs.go

```diff
@@ -0,0 +1,458 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package otelaccesslogging
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	envoy_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	envoy_extensions_access_loggers_grpc_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/grpc/v3"
+	envoy_upstreams_http_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/upstreams/http/v3"
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/api"
+	cmn "github.com/hashicorp/consul/envoyextensions/extensioncommon"
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+const (
+	LocalAccessLogClusterName = "local_access_log"
+
+	localhost     = "localhost"
+	localhostIPv4 = "127.0.0.1"
+	localhostIPv6 = "::1"
+)
+
+type AccessLog struct {
+	CommonConfig       *CommonConfig
+	Body               interface{}
+	Attributes         map[string]interface{}
+	ResourceAttributes map[string]interface{}
+}
+
+func (a *AccessLog) normalize(listenerType string) error {
+	if a.CommonConfig == nil {
+		return fmt.Errorf("missing CommonConfig")
+	}
+
+	return a.CommonConfig.normalize(listenerType)
+}
+
+func (a *AccessLog) validate(listenerType string) error {
+	if err := a.normalize(listenerType); err != nil {
+		return err
+	}
+
+	return a.CommonConfig.validate(listenerType)
+}
+
+type CommonConfig struct {
+	LogName                 string
+	GrpcService             *GrpcService
+	BufferFlushInterval     *time.Duration
+	BufferSizeBytes         uint32
+	FilterStateObjectsToLog []string
+	RetryPolicy             *RetryPolicy
+}
+
+func (c *CommonConfig) normalize(listenerType string) error {
+	if c.GrpcService != nil {
+		c.GrpcService.normalize()
+	} else {
+		return fmt.Errorf("missing GrpcService")
+	}
+
+	if c.RetryPolicy != nil {
+		c.RetryPolicy.normalize()
+	}
+
+	if c.LogName == "" {
+		c.LogName = listenerType
+	}
+
+	return nil
+}
+
+func (c *CommonConfig) validate(listenerType string) error {
+	if c == nil {
+		return nil
+	}
+
+	c.normalize(listenerType)
```

Comment:
I don't think the `listenerType` arg is needed for `validate` because  `CommonConfig.normalize()` is called from `AccessLog.normalize()`. The `normalize()` and `validate()` funcs should not take any args. Eventually I would like these to interface methods.